feat: use LPDAAC specific credentials for downloading

ceholden opened the pull request:
This PR adds support for defining a LPDAAC-specific set of AWS credentials that will be used for downloading imagery. This PR is needed to support using the `s3credentials` endpoint for accessing HLS outputs in lieu of having our container's IAM policy be allowed access directly.

I grabbed the environment variables via `os.getenv` because we might want to revert using specific S3 credentials and fall back to the instance role without changing this container. We can do this by no longer providing these environment variables from the `hls-vi-historical-orchestration` code.

chuckwondo reviewed (changes requested):
Thanks Chris! Would you mind adding a bit of information to the README about where/how to obtain appropriate credentials? Perhaps add something about hitting the appropriate s3credentials endpoint and the need for an EDL account in order to get a successful response.

Perhaps this will be revealed to me when I look at your other PR, but is the idea that the orchestration infra would hit that endpoint and populate these env vars on the container. launch? If so, I suspect somewhere on the orchestration infra you'd be setting EDL creds in order to be able to hit the s3credentials endpoint.

ceholden commented:
Great idea @chuckwondo, it would definitely be nice to make a better link between this container and the orchestration repo.

>Thanks Chris! Would you mind adding a bit of information to the README about where/how to obtain appropriate credentials? Perhaps add something about hitting the appropriate s3credentials endpoint and the need for an EDL account in order to get a successful response.

I added a note next to the `LPDAAC_*` variable references in the README about where these come from when the container is executed via `hls-vi-historical-orchestration`. Happy to iterate on the lingo or where it's best to put in the README, but hopefully that makes sense without being too verbose. I'm planning on writing up some docs on the orchestration system that might be good to link to when they're ready.

>Perhaps this will be revealed to me when I look at your other PR, but is the idea that the orchestration infra would hit that endpoint and populate these env vars on the container launch? If so, I suspect somewhere on the orchestration infra you'd be setting EDL creds in order to be able to hit the s3credentials endpoint.

I refactored the S3 credential fetching into a separate tiny service (Lambda running on a schedule), but generally +1. The credentials are automagically injected into the container as environment variables, but done in a safe way using the AWS Batch <> SecretsManager integration ([see docs](https://docs.aws.amazon.com/batch/latest/userguide/specifying-sensitive-data-secrets.html)). This integration is nice because the AWS Batch Job doesn't show the credentials when describing the job, just a reference that the container will be provided various "secrets" that map into environment variables.

[This PR has the implementation](https://github.com/NASA-IMPACT/hls-vi-historical-orchestration/pull/6), but so far I've just followed when `edl-credential-rotation` did and used a username/password. We could use a EDL token but then we would also need another Lambda to rotate for the EDL token so we can also refresh the S3 credentials. That might be a good addition though since it sounds like it'll reduce some load on NASA user/pass systems and better isolate us from any outages of the user/pass based login (while also saving us a some fractional seconds from the redirect)

chuckwondo commented on README.md:
```diff
@@ -30,6 +30,19 @@ following environment variables must/can be set:
+    as environment variables when running the container.
+    Since these S3 credentials expire after 1 hour, a scheduled Lambda function refreshes
+    the value of the secret with new credentials provided by the LPDAAC `/s3credentials`
+    endpoint.
```
```suggestion
- LPDAAC AWS Credentials (_optional_): AWS credentials for accessing LPDAAC
  data buckets may be provided.

  - `LPDAAC_ACCESS_KEY_ID`
  - `LPDAAC_SECRET_ACCESS_KEY`
  - `LPDAAC_SESSION_TOKEN`

  Values are obtained via a request to the [LPDAAC S3 credentials endpoint].
  Such a request requires a [NASA Earthdata Login] account.

  These are optional in accounts that have been granted direct read permission
  to the LPDAAC protected bucket.

  **NOTE**: The [hls-vi-historical-orchestration] system injects these
  environment variables into this container when running AWS Batch jobs.
```

chuckwondo commented on README.md:
Please add these to the bottom of the file for the new refs in the my other comment:

```
[hls-vi-historical-orchestration]:
  https://github.com/NASA-IMPACT/hls-vi-historical-orchestration
[LPDAAC S3 credentials endpoint]:
  https://data.lpdaac.earthdatacloud.nasa.gov/s3credentials
[NASA Earthdata Login]:
  https://urs.earthdata.nasa.gov/
```

ceholden replied:
Got them! Thanks Chuck, the edits you suggested look great
